lib/Sema/CSDiag.cpp

```diff
@@ -7200,18 +7200,19 @@ bool FailureDiagnosis::visitBindOptionalExpr(BindOptionalExpr *BOE) {
 }
 
 bool FailureDiagnosis::visitIfExpr(IfExpr *IE) {
-  auto typeCheckClauseExpr = [&](Expr *clause) -> Expr * {
-    return typeCheckChildIndependently(clause, Type(), CTP_Unused, TCCOptions(),
-                                       nullptr, false);
+  auto typeCheckClauseExpr = [&](Expr *clause, ContextualTypePurpose convertPurpose) -> Expr * {
```

Comment:
Please pass both type and purpose as parameters (you can have type default to Type() and purpose to CTP_Unused), I am trying to reduce usage of CS.getContextualType in places like this, it could be computed right before the call.